src/components/shared/LessonSide/LevelSelector.tsx

```diff
@@ -1,5 +1,31 @@
-function LevelSelector(): JSX.Element {
-  return <div>Level Selector</div>;
+import '../../../styles/LessonSide.scss';
+
+interface IProps {
+  items: number;
+  current: number;
+}
+
+function LevelSelector({ items, current }: IProps): JSX.Element {
+  const levelList = () => {
+    const list = [];
+    for (let item = 1; item <= items; item++) {
+      let styleClasses = 'level-button other-level';
+      if (item == current) {
+        styleClasses = 'level-button current-level';
+      }
+      list.push(<div className={styleClasses}>{item}</div>);
+      if (item != items) {
+        list.push(<div className="level-connector"></div>);
+      }
+    }
+    return list;
+  };
+
+  return (
+    <div className="selector-box">
+      <div className="level-selector">{levelList()}</div>
```

Comment:
This is what I mean by map :) I'd potentially add a couple of comments so it's easier to understand.  LMK if you want me to explain
1. (bool) ? (true branch) : (false branch)
2. `map` (functional programming, others include `filter`, `forEach`, etc. `map` is most commonly used in React work)
```suggestion
      <div className="level-selector">
        {Array(items * 2 - 1)
          .fill(0)
          .map((_, i) => {
            const level = i/2 + 1;
            return i % 2 === 1 ? (
              <div className="level-connector"></div>
            ) : (
              <div
                className={'level-button' + (level === items ? ' current-level' : '')}
              >
                {level}
              </div>
            );
          })}
      </div>
```

Reply:
Thank you for the detailed feedback! I see how this is a much better way of doing things, I'll do things in the same manner from now!